crates/sui-framework/packages/deepbook/tests/order_query_tests.move

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[test_only]
+module deepbook::order_query_tests {
+    use std::option;
+    use std::option::{none, some, Option};
+    use std::vector;
+    use sui::clock;
+    use deepbook::order_query;
+    use deepbook::order_query::iter_bids;
+    use deepbook::custodian_v2;
+    use deepbook::custodian_v2::{AccountCap, account_owner};
+    use sui::clock::Clock;
+    use sui::coin::mint_for_testing;
+    use deepbook::clob_v2;
+    use sui::sui::SUI;
+    use deepbook::clob_v2::{setup_test, USD, mint_account_cap_transfer, Pool};
+    use sui::test_scenario;
+    use sui::test_scenario::{next_tx, end, ctx, Scenario};
+
+    const CLIENT_ID_ALICE: u64 = 0;
+    const FLOAT_SCALING: u64 = 1000000000;
+    const CANCEL_OLDEST: u8 = 0;
+    const TIMESTAMP_INF: u64 = ((1u128 << 64 - 1) as u64);
+
+    const OWNER: address = @0xf;
+    const ALICE: address = @0xBEEF;
+    const BOB: address = @0x1337;
+
+    #[test]
+    fun test_order_query_pagination() {
+        let scenario = prepare_scenario();
+        add_orders(200, TIMESTAMP_INF, none(), &mut scenario);
+        let pool = test_scenario::take_shared<Pool<SUI, USD>>(&mut scenario);
+        let page1 = iter_bids(&pool, none(), none(), none(), none());
+        assert!(vector::length(order_query::orders(&page1)) == 100, 0);
+        assert!(order_query::has_next_page(&page1), 0);
```

Comment:
does this pagination change if we insert orders in the middle of paginating?

Reply:
the orders are ordered by (tick, order id), so adding a order will change the page, however user can use the max_id filter to ignore any new additions.